frontend/src/components/Chat/index.tsx

```diff
@@ -0,0 +1,201 @@
+      }
+    );
+
+    return () => {
+      communicationSocket.off(CommunicationEvents.USER_JOINED);
+      communicationSocket.off(CommunicationEvents.TEXT_MESSAGE_RECEIVED);
+    };
+  }, []);
+
+  useEffect(() => {
+    endOfMessagesRef.current?.scrollIntoView({ behavior: "smooth" });
+  }, [messages]);
+
+  return (
+    <>
+      <Box
+        sx={{
+          flex: 1,
+          overflowY: "auto",
+          padding: 2,
+        }}
+      >
+        {messages.map((msg, id) =>
+          msg.type === "bot_generated" ? (
+            <Box
+              key={id}
+              sx={(theme) => ({
+                display: "flex",
+                justifyContent: "center",
+                margin: theme.spacing(1, 0),
+              })}
+            >
+              <Typography
+                sx={(theme) => ({
+                  width: "fit-content",
+                  color: theme.palette.secondary.contrastText,
+                  background: theme.palette.secondary.main,
+                  padding: theme.spacing(0.5, 1),
+                  borderRadius: theme.spacing(1),
+                  fontSize: "12px",
+                  fontWeight: "bold",
+                })}
+              >
+                {msg.message}
+              </Typography>
+            </Box>
+          ) : msg.from === user?.username ? (
+            <Box
+              key={id}
+              sx={(theme) => ({
+                display: "flex",
+                justifyContent: "flex-end",
+                marginTop: theme.spacing(1),
+              })}
+            >
+              <Typography
+                sx={(theme) => ({
+                  background: theme.palette.primary.main,
+                  padding: theme.spacing(1, 2),
+                  borderRadius: theme.spacing(2),
+                  maxWidth: "80%",
+                })}
+              >
+                {msg.message}
+              </Typography>
+            </Box>
+          ) : (
+            <Box
+              key={id}
+              sx={(theme) => ({
+                display: "flex",
+                justifyContent: "flex-start",
+                marginTop: theme.spacing(1),
+              })}
+            >
+              <Typography
+                sx={(theme) => ({
+                  background: theme.palette.secondary.main,
+                  padding: theme.spacing(1, 2),
+                  borderRadius: theme.spacing(2),
+                  maxWidth: "80%",
+                })}
+              >
+                {msg.message}
+              </Typography>
+            </Box>
+          )
+        )}
+      </Box>
+      <div ref={endOfMessagesRef} />
+      <TextField
+        placeholder="Type message..."
+        margin="none"
+        multiline
+        fullWidth
+        value={inputValue}
+        onChange={(e) => setInputValue(e.target.value)}
+        onKeyDown={(e) => {
+          if (e.key === "Enter" && inputValue !== "") {
```

Comment:
should trim the inputValue also? cos i can submit a message with just a new line